### DISCUSSION TITLE
git@github.com: Permission denied (publickey).

### DISCUSSION THREAD

**Discussion by sth-ifsb** [community/community#51502] [Actions] [answered]
### Select Topic Area

Question

### Body

Under Debian I successfully cloned my private repo webapp via ssh
I forked master with dev branch and set dev as default
I wanted now to clone master to a prod environment. 
I needed to regenerate a private and public ssh key, I put public in ssh key github account. fingerprint confirm this is the good key.
Unfortunately, I cannot clone my master repo to prod because of git@github.com: Permission denied (publickey).
I tried to do again and again the steps, creating private key, public key, copying-pasting again and again.
`ssh -vT git@github.com`
says it's ok : Hi xxx! You've successfully authenticated, but GitHub does not provide shell access.

`sudo git clone -b master git@github.com:xxx/webapp.git` doesn't work

I checked my .ssh, config, etc.
What do you need to help me ?
Thank you in advance

**Comment by airtower-luna** [comment, marked as answer]
Do not use `sudo`. Sudo makes the command run as `root` (also known as the "super user"), which means it'd use the keys set up for the `root` user, not yours.

Generally you should *never* run Git as `root` (via `sudo` or otherwise). It doesn't need extra rights, so granting them anyway is an unnecessary security risk. :warning:

**Reply by sth-ifsb** [reply]
I indeed realized that by the of the day and solved this issue without sudo.
Was so obvious :(
Thank you so much !
